components/post_view/post_aria_label_div.tsx

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useRef} from 'react';
+import {useSelector} from 'react-redux';
+import {useIntl} from 'react-intl';
+
+import {Post} from 'mattermost-redux/types/posts';
+
+import {GlobalState} from 'types/store';
+
+import {makeCreateAriaLabelForPost} from 'utils/post_utils';
+
+type Props = React.HTMLProps<HTMLDivElement> & {
+    labelPrefix?: string;
+    post: Post;
+}
+
+const PostAriaLabelDiv = React.forwardRef((props: Props, ref: React.Ref<HTMLDivElement>) => {
+    const {post, labelPrefix, ...otherProps} = props;
+
+    const intl = useIntl();
+
+    const createAriaLabelForPost = useRef(makeCreateAriaLabelForPost());
+    let ariaLabel = useSelector<GlobalState, string>((state) => createAriaLabelForPost.current(state, post)(intl));
+    if (labelPrefix) {
+        ariaLabel = labelPrefix + ariaLabel;
+    }
+
+    return (
+        <div
+            ref={ref}
+            aria-label={ariaLabel}
+            {...otherProps}
```

Comment:
Does this implicitly put `props.children` inside of the `div`? If this is the case, an we explicitly put them inside of the `div` to be clear about the role of this component?

Reply:
Yep, it does pass children through. I can do that explicitly though.